src/tm-threads.c

```diff
@@ -1713,16 +1766,32 @@ TmEcode TmThreadSpawn(ThreadVars *tv)
         return TM_ECODE_FAILED;
     }
 
-    /* Initialize and set thread detached attribute */
-    pthread_attr_init(&attr);
+#ifdef HAVE_DPDK
+    /* only DPDK worker threads are launched using rte_eal_remote_launch */
+    if (tv->lcore_id != LCORE_ID_ANY) {
+        struct dpdk_callback cb = { .func = tv->tm_func, .tv = tv };
 
-    pthread_attr_setdetachstate(&attr, PTHREAD_CREATE_JOINABLE);
+        int rc = rte_eal_remote_launch(&TmThreadSpawnDpdkThread, (void *)&cb, tv->lcore_id);
```

Comment:
there is no mandate to launch dpdk processing function with rte_ela_launch or rte_eal_remote_launch. Please rework

Reply:
reworked to pthreads only.